aten/src/ATen/native/cuda/DistributionTemplates.h

```diff
@@ -161,7 +167,101 @@ void distribution_nullary_kernel(at::TensorIterator& iter,
   AT_CUDA_CHECK(cudaGetLastError());
 }
 
+// Binary kernel
+template <typename func_t, typename inp_offset_calc_t, typename out_offset_calc_t>
+__global__ void distribution_binary_elementwise_kernel(
+    int numel,
+    func_t f,
+    std::pair<uint64_t, uint64_t> seeds,
+    typename ternary_function_traits<func_t>::result_type *output_data,
+    const typename ternary_function_traits<func_t>::arg2_t *input_data_1,
+    const typename ternary_function_traits<func_t>::arg3_t *input_data_2,
+    inp_offset_calc_t inp_calc,
+    out_offset_calc_t out_calc) {
+  using input_t_1 = typename ternary_function_traits<func_t>::arg2_t;
+  using input_t_2 = typename ternary_function_traits<func_t>::arg3_t;
+
+  input_t_1 inputs_1[THREAD_WORK_SIZE];
+  input_t_2 inputs_2[THREAD_WORK_SIZE];
+
+  int base_index = BLOCK_WORK_SIZE * blockIdx.x;
+  int remaining = std::min<int>(numel - base_index, BLOCK_WORK_SIZE);
+
+  curandStatePhilox4_32_10_t state;
+  curand_init(seeds.first, blockIdx.x * blockDim.x + threadIdx.x, seeds.second, &state);
+
+  // load data into registers
+  int thread_idx = threadIdx.x;
+  #pragma unroll
+  for (int i = 0; i < THREAD_WORK_SIZE; i++) {
+    if (thread_idx >= remaining) {
+      break;
+    }
+    int input_idx = thread_idx + base_index;
+    auto offsets = inp_calc.get(input_idx);
+    inputs_1[i] = input_data_1[offsets[0]];
+    inputs_2[i] = input_data_2[offsets[1]];
+
+    thread_idx += num_threads;
+  }
```

Comment:
Did you try if changing the `THREAD_WORK_SIZE` into `thread_work_size` work? Ideally, it should be `thread_work_size`, but unfortunately, sometimes CUDA would complain that some variable is not available on device, although declared as `constexpr`, that's why there is a `THREAD_WORK_SIZE`, but `thread_work_size` is still preferred when available...

Reply:
No I didn't. Understood, trying that.